x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderServiceTests.java

```diff
@@ -521,25 +520,26 @@ public void testUnmovableSize() {
         stateBuilder.metadata(metaBuilder);
         ClusterState clusterState = stateBuilder.build();
 
-        Set<ShardRouting> shards = IntStream.range(0, between(1, 10))
+        var shards = IntStream.range(0, between(1, 10))
             .mapToObj(i -> Tuple.tuple(new ShardId(indexMetadata.getIndex(), randomInt(10)), randomBoolean()))
             .distinct()
             .map(t -> TestShardRouting.newShardRouting(t.v1(), nodeId, t.v2(), ShardRoutingState.STARTED))
-            .collect(Collectors.toSet());
+            .toList();
```

Comment:
This change is required to make test reproducible.

ShardRouting hashCode relies on AllocationId. In this case allocation is generated randomly at https://github.com/elastic/elasticsearch/blob/e1c96c4837482649391d5714cdd686f3cd8086e3/test/framework/src/main/java/org/elasticsearch/cluster/routing/TestShardRouting.java#L270-L276 
without test random source. This result in a different hash order of ShardRoutings in the set. 
Later in the test we rely on the test order when picking random shard with `randomFrom(shards)`.